source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -890,6 +895,91 @@ def _filter_similar_journeys(journey_pairs_pool, request, *similar_journey_gener
             )
 
 
+def filter_odt_journeys(journeys, request):
+    clockwise = request.get('clockwise', True)
+    debug = request.get('debug', False)
+    journeys_list = [j for j in journeys]
+    if clockwise:
+        return _filter_odt_journeys_clockwise(journeys_list, debug)
+    else:
+        return _filter_odt_journeys_counter_clockwise(journeys_list, debug)
+
+
+def _filter_odt_journeys_clockwise(journeys, debug):
+    """
+    eliminates a journey that uses On Demand Transport if there is a public transport journey
+    that arrive earlier
+    """
+    # let's find the the earliest arrival time among public transport journeys
+    earliest_arrival_pt_journey = None
+    for journey in journeys:
```

Comment:
I think what you need here is to find the earliest arrival_time?

Reply:
I updated my suggestion

Reply:
done in https://github.com/hove-io/navitia/pull/3772/commits/8509edba823679ed05ea524b37319e2a2d5cec5e